hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieInstantTimeGenerator.java

```diff
@@ -65,6 +71,12 @@ public static Date parseInstantTime(String timestamp) {
       if (timestamp.equals(ALL_ZERO_TIMESTAMP)) {
         return new Date(0);
       }
+      // compaction and cleaning in metadata has special format. handling it here.
+      if (timestamp.length() == MILLIS_INSTANT_ID_LENGTH) {
```

Comment:
btw this is not just millisecond format. its going to add `001` to the millisecond format once you land that

Reply:
yes, I will rework this again once with the other patch. plan is to land this one and then rework with ms granularity patch. other patch has lot of conflicts to be fixed anyways.